src/edu/umass/cs/gnscommon/utils/CanonicalJSON.java

```diff
@@ -118,7 +118,16 @@ private static String renderSimpleCanonicalJSON(Object x) {
           throw new JSONException("Bad value from toJSONString: " + object);
         }
         if (x instanceof Number) {
-          return JSONObject.numberToString((Number) x);
+          if (x instanceof Double || x instanceof Float) {
+            // This is a workaround for the issue described in MOB-1103 where the iOS and 
+            // potentially other clients have different printed representations for floating point numbers.
+            // This effectively removes floats from the canonical json string that is returned.
+            // A similar patch will be made on the ios client.
+            return "float";
```

Comment:
That means, I can modify the contents of the "message" (say http query parameters) while still retaining the same signature, correct? This doesn't seem quite right to me. I'll wait for others to review this.